Source/Core/Common/Network.h

```diff
@@ -99,8 +115,124 @@ struct UDPHeader
 };
 static_assert(sizeof(UDPHeader) == UDPHeader::SIZE);
 
+#pragma pack(push, 1)
+struct ARPHeader
+{
+  ARPHeader();
+  ARPHeader(u32 from_ip, MACAddress from_mac, u32 to_ip, MACAddress to_mac);
+  u16 Size() const;
+
+  static constexpr std::size_t SIZE = 28;
+
+  u16 hardware_type = 0;
+  u16 protocol_type = 0;
+  u8 hardware_size = 0;
+  u8 protocol_size = 0;
+  u16 opcode = 0;
+  MACAddress sender_address{};
+  u32 sender_ip = 0;
+  MACAddress targer_address{};
+  u32 target_ip = 0;
+};
+static_assert(sizeof(ARPHeader) == ARPHeader::SIZE);
+#pragma pack(pop)
+
+struct DHCPBody
+{
+  DHCPBody();
+  DHCPBody(u32 transaction, MACAddress client_address, u32 new_ip, u32 serv_ip);
+  DHCPBody(std::vector<u8> data);
+  bool AddDHCPOption(u8 size, u8 fnc, const std::vector<u8>& params);
+  static constexpr std::size_t SIZE = 540;
```

Comment:
I don't think that's the correct value. I found `548` on my own which matches the RFC 1531 specification:
> This brings the minimum size of a DHCP message up to 576 octets

576 = IP header (20) + UDP header (8) + 548.

Reply:
this size doesnt include the headers, its a payload, the actual used size is 300 bytes. I should probably make a builder/reader like the udp/tcp stuff